tests/fuzzer/fuzzer_v2.py

```diff
@@ -0,0 +1,629 @@
+            print("ignore: no path found")
+            return
+        _make_error(_ErrorType.STDCM, r, infra_name, stdcm_payload=stdcm_payload, prelude=prelude)
+    print("test PASSED")
+
+
+def _make_stdcm_payload(path: List[Tuple[str, float]], rolling_stock: int) -> Dict:
+    """
+    Creates a payload for an STDCM request
+    """
+    res = {
+        "rolling_stock_id": rolling_stock,
+        "start_time": _make_random_time(),
+        "maximum_departure_delay": random.randint(0, 3_600_000 * 4),
+        "maximum_run_time": random.randint(3_600_000 * 5, 3_600_000 * 10),
+        "time_gap_before": random.randint(0, 600_000),
+        "time_gap_after": random.randint(0, 600_000),
+        "steps": [_convert_stop_stdcm(stop) for stop in path],
+        "comfort": "STANDARD",
+        "margin": "None",
+    }
+    res["steps"][-1]["duration"] = 1  # Force a stop at the end
+    allowance_value = _make_random_margin_value()
+    if random.randint(0, 2) == 0:
+        res["standard_allowance"] = allowance_value
+    return res
+
+
+@cache
+def _get_rolling_stock(editoast_url: str, rolling_stock_name: str) -> _RollingStock:
+    """
+    Returns the ID corresponding to the rolling stock name, if available.
+    :param editoast_url: Api url
+    :param rolling_stock_name: name of the rolling stock
+    :return: ID the rolling stock
+    """
+    return _RollingStock(rolling_stock_name, conftest.get_rolling_stock(editoast_url, rolling_stock_name))
+
+
+def _get_random_rolling_stock(editoast_url: str) -> _RollingStock:
+    """
+    Returns a random rolling stock ID
+    :param editoast_url: Api url
+    :return: ID of a valid rolling stock
+    """
+    # TODO: we may want to check more pages, for more randomness
+    r = _get_with_timeout(editoast_url + "light_rolling_stock/")
+    if r.status_code // 100 != 2:
+        raise RuntimeError(f"Rolling stock error {r.status_code}: {r.content}")
+    stocks = r.json()["results"]
+    rolling_stock = random.choice(stocks)
+    return _RollingStock(rolling_stock["name"], rolling_stock["id"])
+
+
+def _make_graph(editoast_url: str, infra: int) -> _InfraGraph:
+    """
+    Makes a graph from the infra
+    :param editoast_url: editoast url
+    :param infra: infra id
+    """
+    url = editoast_url + f"infra/{infra}/railjson/"
+    r = _get_with_timeout(url)
+    infra = r.json()
+    graph = _InfraGraph(infra)
+
+    for track in infra["track_sections"]:
+        graph.tracks[track["id"]] = track
+
+    for switch in infra["switches"]:
+        switch_type = builtin_node_types()[switch["switch_type"]]
+        for group in switch_type["groups"].values():
+            for pair in group:
+                src = _TrackEndpoint.from_dict(switch["ports"][pair["src"]])
+                dst = _TrackEndpoint.from_dict(switch["ports"][pair["dst"]])
+                graph.link(src, dst)
+    return graph
+
+
+def _random_set_element(s: Iterable):
```

Comment:
nit: perhaps we could annotate the output thanks to TypeVar https://docs.python.org/3/library/typing.html#generics